Add LMATH

charlesknightsbridge opened the pull request:
LMATH is required by regress

charlesknightsbridge committed:
Add LMATH

torstenchr commented:
A few things before this Pull Request is ready to merge:

* All backup folders and files must be removed and preferable these folders should be added to the .gitignore file (if not already present.)
* Each component needs a pom.xml file describing what to do when compiling in the github action. You can take a look at LUIPack folder for inspiration on how this should be setup.
* The master pom.xml (in the root folder) should add this new folder as a dependency ("modules" section)
* The assembly.xml file must be extended to include the compiled files from the new component.
* The config folder has a configuration file that must be updated with the new paths of the LMath resulting files

charlesknightsbridge committed:
Removed LMATH backup folders

charlesknightsbridge committed:
Added / modified files for assembly

charlesknightsbridge committed:
Fixed type in LMath/pom.xml

charlesknightsbridge committed:
Fixed another typo in pom.xml

charlesknightsbridge committed:
Correct ModelVersion for LMath

charlesknightsbridge committed:
Correct version of Maven

charlesknightsbridge committed:
Correct version for components-parent

Also fixed typo in lmSpecRegress module

charlesknightsbridge committed:
The pom-pom routine

Rename lmMathUtils to lmMathUtil
create pom.xml and .fpc files for all subcomponents of lmath

charlesknightsbridge committed:
Correct case of lmOptimum

charlesknightsbridge committed:
Change order of lmath modules

Compilation order is important1

charlesknightsbridge committed:
set compiler mode -Mfpc

For all of LMath

torstenchr committed:
Fixed compilation errors with LMath

charlesknightsbridge commented:
Is this ready? I don't know what the 'Compile the components' is waiting for

charlesknightsbridge commented:
@torstenchr 
My reading of the github help leads me to believe that the "Compile the components" got hung up because the build components steps were running concurrently and the compile step could not get started. I wonder if this will work if we force github to run the steps sequentially. In any case, I cannot change this behaviour.
